### DISCUSSION TITLE
Convertire messaggi Tuya in MQTT per HA tramite Node-Red

### DISCUSSION THREAD

**Discussion by Topolynx** [indomus/forum#293] [HUB: Home Assistant]
Ciao a tutti, anzitutto vi chiedo scusa in anticipo dal momento che ho aperto più o meno sempre discussioni macchinose.
Sto cercando di migrare uno scaldino smart della Bimar con chip Tuya dall'integrazione Tuya di Home Assistant al protocollo MQTT attraverso Node-Red, e vi spiego perché prima che mi prendiate per cretino.
Con l'integrazione Tuya standard, HA vede lo scaldino come climate, ma non lo controlla in toto, ad esempio non mi è possibile controllare l'oscillazione della testa né la potenza di riscaldamento. In aggiunta, c'è un lag tra i comandi e gli aggiornamenti di stato pazzesco, anche 5 minuti. Utilizzo già l'integrazione LocalTuya per altri smart plugs, però le entità climate non sono ancora supportate. Ho trovato una via di uscita con Node-Red ed un nodo che controlla tutte le funzioni di questo scaldino sempre in locale e con ritardo zero. Adesso si tratta di interfacciare Node-Red e HA. Attraverso i nodi per HA posso solo creare o un sensore, o un sensore binario o uno switch, quindi MQTT mi sembra l'ultima sponda per raggiungere i miei scopi. Da dove devo partire? Devo creare anzitutto un dispositivo climate nel configuration.yaml?
Per cortesia aiutatemi se siete pratici.

**Comment by xyzmauro** [comment]
Ciao, mai fatto ma vado ad istinto, usa questo e da NodeRed poi pubblichi i payload corretti sui topic che definisci tu
https://www.home-assistant.io/integrations/climate.mqtt/

**Comment by indomus** [comment]
Ti basta usare [questa guida](https://indomus.it/progetti/trasformare-piu-elementi-mqtt-in-un-unico-dispositivo-virtuale-tramite-node-red/). Il concetto è più o meno lo stesso, semplicemente invece di usare un alarm control panel (come fatto [qui](https://indomus.it/progetti/domotizzare-un-sistema-dallarme-tradizionale-con-mqtt-e-node-red-parte-1/)) devi usare climate.mqtt come consigliato a xyzmauro

**Comment by Topolynx** [comment]
Grazie mille :)

**Reply by monkeypr00f** [reply]
Posso chiederti che componente nodered hai usato? Richiede il device Id e local key? Grazie
